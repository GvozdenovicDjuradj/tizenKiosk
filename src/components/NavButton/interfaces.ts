import { ViewStyle } from "react-native"
import { ButtonProps } from "../Button"

export interface Styles {
  navButton: ViewStyle
  navButtonLeft: ViewStyle
  navButtonRight: ViewStyle
  triangleLeft: ViewStyle
  triangleRight: ViewStyle
  view: ViewStyle
}

export interface NavButtonProps extends ButtonProps {
  buttonColor?: string
  privacyPolicyButtonDefaultText?: string
}

export interface NavButtonNextState {
  isDisabled: boolean
}
