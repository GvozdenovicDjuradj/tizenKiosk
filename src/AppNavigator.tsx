import React from "react"
import { BackHandler } from "react-native"
import { connect } from "react-redux"
import { createStackNavigator, NavigationInjectedProps } from "react-navigation"
import { AppScreens, RootState } from "./interfaces"
import { selectors } from "./utils"
import ConfirmationScreen from "./components/confirmationScreen"
import CustomerScreen from "./components/customerScreen"
import Home from "./components/Home"
import ServiceSelectionScreen from "./components/serviceSelection"
import SubserviceSelectionScreen from "./components/serviceSelection/SubServiceSelection"
import ServiceInfoScreen from "./components/serviceSelection/ServiceInfoScreen"
import NoConnectionModal from "./components/NoConnectionModal"
import CheckInScreen from "./components/checkInScreen"
import CheckInConfirmation from "./components/checkInConfirmationScreen"
import EventCheckInScreen from "./components/eventCheckInScreen"
import EventCheckInConfirmationScreen from "./components/eventCheckInConfirmationScreen"
import KioskClosedScreen from "./components/kioskClosedScreen"
import QuestionsScreen from "./components/QuestionsScreen"
import QueueFullOrNAScreen from "./components/QueueFullOrNAScreen"
import ScreenSaver from "./components/screenSaver"
import { handleBackButton, removeBackButtonHandler } from "./utils/backButton"
import PrivacyPolicyScreen from "./components/privacyPolicyScreen"
import KeepAwake from 'react-native-keep-awake'
import QueueUnderCapacityScreen from "./components/QueueUnderCapacityScreen"

const AppStackNavigator = createStackNavigator({
  [AppScreens.HOME]: { screen: Home },
  [AppScreens.SERVICE_SELECTION]: { screen: ServiceSelectionScreen },
  [AppScreens.SUBSERVICE_SELECTION]: { screen: SubserviceSelectionScreen },
  [AppScreens.SERVICE_INFO]: { screen: ServiceInfoScreen },
  [AppScreens.CUSTOMER_DETAILS]: { screen: CustomerScreen },
  [AppScreens.SCREEN_SAVER]: { screen: ScreenSaver },
  [AppScreens.QUEUE_CONFIRMATION]: {
    screen: ConfirmationScreen,
    navigationOptions: {
      gesturesEnabled: false,
    },
  },
  [AppScreens.QUEUE_FULL_OR_NA]: { screen: QueueFullOrNAScreen },
  [AppScreens.QUEUE_UNDER_OCCUPANCY]: { screen: QueueUnderCapacityScreen },
  [AppScreens.QUESTIONS]: { screen: QuestionsScreen },
  [AppScreens.CHECK_IN]: { screen: CheckInScreen },
  [AppScreens.EVENT_CHECK_IN]: { screen: EventCheckInScreen },
  [AppScreens.EVENT_CHECK_IN_CONFIRMATION]: { screen: EventCheckInConfirmationScreen },
  [AppScreens.NO_CONNECTION_MODAL]: { screen: NoConnectionModal },
  [AppScreens.KIOSK_CLOSED]: { screen: KioskClosedScreen },
  [AppScreens.CHECK_IN_CONFIRMATION]: {
    screen: CheckInConfirmation,
    navigationOptions: {
      gesturesEnabled: false,
    },
  },
  [AppScreens.PRIVACY_POLICY]: { screen: PrivacyPolicyScreen },
}, {
  cardStyle: { backgroundColor: "transparent" },
  headerMode: "none",
  initialRouteName: AppScreens.HOME,
  mode: "card",
})

interface Props extends NavigationInjectedProps {
  loading: boolean,
  hasKioskModeEnable: boolean
}

class AppNavigator extends React.Component<Props> {
  public static router = AppStackNavigator.router

  /**
   * Depending on hasKioskModeEnable props, calling back button will Quit the app or do nothing
   * react navigation back button handler will overwrite that behavior
   */
  public handleBackButtonClick = () => {
   if (!this.props.hasKioskModeEnable) {
     BackHandler.exitApp()
   }
  }

  public componentWillMount() {
    handleBackButton(this.handleBackButtonClick)
  }

  public componentWillUnmount() {
    removeBackButtonHandler(this.handleBackButtonClick)
  }

  public render() {
    const { hasKioskModeEnable } = this.props
    return (
      <React.Fragment>
        { hasKioskModeEnable &&
          <KeepAwake />
        }
        <AppStackNavigator navigation={this.props.navigation} />
      </React.Fragment>
    )
  }
}

const mapStateToProps = (state: RootState) => ({
  hasKioskModeEnable: selectors.isKioskModeEnabled(state)
})

export default connect(mapStateToProps)(AppNavigator)
