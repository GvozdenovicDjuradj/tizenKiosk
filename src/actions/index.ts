export * from "./app"
export * from "./error"
export * from "./kiosk"
export * from "./kioskSettings"
export * from "./printing"
export * from "./navigationActions"
export * from "./screenSaver"
