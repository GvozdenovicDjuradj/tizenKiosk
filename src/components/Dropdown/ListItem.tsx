import React from "react"
import {
  Text,
  View,
} from "react-native"
import { listItem } from "./styles"

interface Props {
  onPress: (item: string) => void;
  selected?: boolean;
  selectedColor?: string;
  title: string;
}

export default (props: Props) => {
  const { onPress, selected, selectedColor, title } = props
  const buttonStyle = [
    listItem.button,
    selected ? [
      listItem.selected,
      { borderColor: selectedColor }
    ] : {}
  ]
  const textStyle = [
    listItem.text,
    selected ? { color: selectedColor } : {}
  ]
  return (
    <View
      onStartShouldSetResponder={() => true}
      onResponderRelease={() => onPress(title)}
      style={buttonStyle}
    >
      <Text style={textStyle}>
        {title}
      </Text>
    </View>
  )
}
