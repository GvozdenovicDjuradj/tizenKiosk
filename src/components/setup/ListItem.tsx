import React, { Component } from "react"
import { Text, View } from "react-native"
import { font } from "./styles"

export interface RegistrationStepItem {
  key: string;
  title: string;
  children?: RegistrationStepItem[];
}

interface Props {
  item: RegistrationStepItem;
}

enum BulletType {
  ONE = "\u2022",
  TWO = "\t\t\u25E6",
}

export default class ListItem extends Component<Props> {
  public render(): JSX.Element {
    const { item } = this.props
    let children
    if (item.children && item.children.length) {
      children = this.renderChildren(item.children)
    }
    return (
      <View>
        <Text style={font}>{BulletType.ONE} {item.title}</Text>
        {children}
      </View>
    )
  }

  private renderChildren(children: RegistrationStepItem[]) {
    const items = children.map((child: RegistrationStepItem, i: number) => (
      <Text key={i} style={font}>{BulletType.TWO} {child.title}</Text>
    ))
    return (
      <View>{items}</View>
    )
  }
}
