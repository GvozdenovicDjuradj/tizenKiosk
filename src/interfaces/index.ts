export * from "./answers"
export * from "./app"
export * from "./appState"
export * from "./checkIn"
export * from "./customerInQueue"
export * from "./eventCheckIn"
export * from "./kiosk"
export * from "./modal"
export * from "./navigation"
export * from "./product"
export * from "./question"
export * from "./routes"
export * from "./utils"
export * from "./validation"
export * from "./venue"
export * from "./screenSaver"

import { AppState } from "./app"
import { CheckInState } from "./checkIn"
import { EventCheckInState } from "./eventCheckIn"
import { KioskSettingsForm } from "../reducers/kioskSettings"
import { KioskState } from "./kiosk"
import { ModalState } from "./modal"
import { QuestionAnswer } from "./answers"
import { QuestionsReducer } from "./question"
import { ValidationState } from "./validation"
import { PrintingState } from "./printing"
import { PrinterState } from "../printer/reducer"
import { ScreenSaverState } from "./screenSaver"

export interface Flags {
  [key: string]: any
}

export type ValidatorFn = (text: string, ...args: any[]) => { error: string }

export interface RootState {
  answers: QuestionAnswer[];
  app: AppState;
  checkIn: CheckInState;
  error: string;
  eventCheckIn: EventCheckInState;
  kiosk: KioskState;
  kioskSettings: KioskSettingsForm;
  modal: ModalState;
  printer: PrinterState;
  printing: PrintingState;
  questions: QuestionsReducer;
  validation: ValidationState;
  screenSaver?: ScreenSaverState;
}
