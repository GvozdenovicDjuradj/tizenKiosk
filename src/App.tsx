import React, { Component } from "react"
import { Store } from "redux"
import { Provider } from "react-redux"
import { StatusBar, View, PermissionsAndroid, Platform } from "react-native"
import { PersistGate } from "redux-persist/integration/react"
import { Navigator as NavigatorType, setNavigator } from "./navigation"
import Navigator from "./routes"
import { applicationStart, initApp as initAppActionCreator, stopApp } from "./actions/app"
import { RootState } from "./interfaces"
import { persistor, store } from "./store"
import { setup } from "./utils/localeString"
import Modals from "./modals"
import Printing from "./components/Printing"
import Ticket from "./printer/containers/Ticket"
import Notification from "./components/Notification"
import initSentry from "./initSentry"
import { getKioskIdAndDomainFromUrl } from "./actions"
import FadeNotification from "./printer/containers/FadeNotification"

if (Platform.OS !== "web") {
    initSentry()
}

setup(store)

if (Platform.OS === "web") {
    setTimeout(() => {
        store.dispatch(getKioskIdAndDomainFromUrl())
    }, 0)
}

setTimeout(() => store.dispatch(applicationStart()))

const start = (store$: Store<RootState>, navigator: NavigatorType) => {
    if (navigator) {
        setNavigator(navigator)
        store$.dispatch(initAppActionCreator())

    } else {
        store$.dispatch(stopApp())
    }
}

export default class App extends Component {
    public async componentDidMount() {
        if (Platform.OS === "android") {
            // TODO - This should be re-prompted in settings later, I will handle this in a later ticket
            await PermissionsAndroid.request(
              PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
              {
                title: "Qudini Kiosk - Printing",
                message: `We require access to read a configuration
                file which stores internal settings for the connected
                printer, in addition we also need to read saved tickets
                which we save to external storage as an image.`,
                buttonNegative: "Decline",
                buttonPositive: "Accept",
              },
            )

            await PermissionsAndroid.request(
              PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
              {
                title: "Qudini Kiosk - Printing",
                message: `We require access to write to a configuration
                  file which stores internal settings for the connected
                  printer, in addition we also need to write saved tickets
                  which we save to external storage as an image.`,
                buttonNegative: "Decline",
                buttonPositive: "Accept",
              },
            )
        }
    }

    public render() {
        return (
            <Provider store={store}>
                <PersistGate persistor={persistor}>
                    <View style={{ flex: 1 }}>
                        <StatusBar hidden/>
                        <Ticket/>
                        <View style={{ flex: 1, width: "100%", height: "100%", overflow: "hidden" }}>
                            <Modals/>
                            <Notification/>
                            <Navigator ref={(navigator: NavigatorType) => start(store, navigator)}/>
                            <Printing />
                        </View>
                        <FadeNotification />
                    </View>
                </PersistGate>
            </Provider>
        )
    }
}
