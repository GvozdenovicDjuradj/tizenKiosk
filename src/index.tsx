import "babel-polyfill"
import { AppRegistry } from "react-native"
import App from "./App"
import app from "../app.json"

AppRegistry.registerComponent(app.name, () => App)

if (window && typeof window === "object" && window.document) {
  AppRegistry.runApplication(app.name, { rootTag: document.getElementById("app") })
}
