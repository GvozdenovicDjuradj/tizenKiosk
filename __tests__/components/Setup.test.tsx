import React from "react"
import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import mockStore from "redux-mock-store"
import Setup from "../../src/components/setup"
import ListItem, { RegistrationStepItem } from "../../src/components/setup/ListItem"
import { AppScreens } from "../../src/interfaces"

describe("Setup component tests", () => {

  it("should render Setup component successfully", () => {
    const state = {
      kiosk: {
        fields: {
          kioskIdentifier: "1",
          printerUrl: "",
          hasPrinter: false,
          url: "",
        }
      },
      printer: {
        connectedPrinters: []
      },
      validation: { [AppScreens.HOME]: { url: { error: [] } } }
    }
    const wrapper = renderer.create(
      <Provider store={mockStore()(state)}>
        <Setup />
      </Provider>
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it("should render Setup ListItem component successfully", () => {
    const item: RegistrationStepItem = {
      key: "1",
      title: "title1",
      children: [{
        key: "1.1",
        title: "title1.1"
      }, {
        key: "1.2",
        title: "title1.2"
      }]
    }
    const wrapper = renderer.create(<ListItem item={item} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

})
