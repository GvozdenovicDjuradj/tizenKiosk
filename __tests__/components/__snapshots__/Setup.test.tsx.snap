// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup component tests should render Setup ListItem component successfully 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "Arial",
        "fontSize": 20,
      }
    }
  >
    •
     
    title1
  </Text>
  <View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Arial",
          "fontSize": 20,
        }
      }
    >
      		◦
       
      title1.1
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Arial",
          "fontSize": 20,
        }
      }
    >
      		◦
       
      title1.2
    </Text>
  </View>
</View>
`;

exports[`Setup component tests should render Setup component successfully 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={false}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F80347",
            "justifyContent": "center",
            "marginTop": 80,
            "padding": 20,
            "width": "100%",
          },
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 55,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {},
          ]
        }
      >
        Kiosk setup
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 3,
            "flexDirection": "row",
            "paddingHorizontal": 50,
            "paddingTop": 40,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 0.6,
            "paddingRight": 15,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginBottom": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 32.5,
                "borderWidth": 7,
                "height": 65,
                "justifyContent": "center",
                "width": 65,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 40,
                  "fontWeight": "bold",
                }
              }
            >
              1
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": 65,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "Arial",
                  "fontSize": 20,
                }
              }
            >
              Add a kiosk to your venue
            </Text>
          </View>
        </View>
        <RCTScrollView
          data={
            Array [
              Object {
                "key": "1",
                "title": "Login to app.qudini.com with a merchant or venue admin account",
              },
              Object {
                "key": "2",
                "title": "Click ‘Settings’ then ‘Kiosk’",
              },
              Object {
                "children": Array [
                  Object {
                    "key": "3.1",
                    "title": "Kiosk description (just for your reference as to the location of the kiosk)",
                  },
                  Object {
                    "key": "3.2",
                    "title": "Select a kiosk design template (defined by your merchant admin)",
                  },
                  Object {
                    "key": "3.3",
                    "title": "Select the products to be used for this kiosk",
                  },
                ],
                "key": "3",
                "title": "Add a kiosk and set the following:",
              },
              Object {
                "key": "4",
                "title": "Press save. This then generates a kiosk ID to be used in step 2.",
              },
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "Arial",
                      "fontSize": 20,
                    }
                  }
                >
                  •
                   
                  Login to app.qudini.com with a merchant or venue admin account
                </Text>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "Arial",
                      "fontSize": 20,
                    }
                  }
                >
                  •
                   
                  Click ‘Settings’ then ‘Kiosk’
                </Text>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "Arial",
                      "fontSize": 20,
                    }
                  }
                >
                  •
                   
                  Add a kiosk and set the following:
                </Text>
                <View>
                  <Text
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "Arial",
                        "fontSize": 20,
                      }
                    }
                  >
                    		◦
                     
                    Kiosk description (just for your reference as to the location of the kiosk)
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "Arial",
                        "fontSize": 20,
                      }
                    }
                  >
                    		◦
                     
                    Select a kiosk design template (defined by your merchant admin)
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "Arial",
                        "fontSize": 20,
                      }
                    }
                  >
                    		◦
                     
                    Select the products to be used for this kiosk
                  </Text>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "Arial",
                      "fontSize": 20,
                    }
                  }
                >
                  •
                   
                  Press save. This then generates a kiosk ID to be used in step 2.
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-end",
              "paddingBottom": 15,
              "paddingRight": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "printer.png",
              }
            }
            style={
              Object {
                "height": 40,
                "resizeMode": "contain",
                "width": 136,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 0.4,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "marginBottom": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 32.5,
                "borderWidth": 7,
                "height": 65,
                "justifyContent": "center",
                "width": 65,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 40,
                  "fontWeight": "bold",
                }
              }
            >
              2
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": 65,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "Arial",
                    "fontSize": 20,
                  },
                  Object {
                    "paddingLeft": 15,
                  },
                ]
              }
            >
              Register your Kiosk
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#dfdfdf",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "width": "100%",
                },
                Object {},
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              dispatch={[Function]}
              name="url"
              onChange={[Function]}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              onSubmitEditing={[Function]}
              placeholder="URL"
              rejectResponderTermination={true}
              returnKeyType="next"
              style={
                Array [
                  Object {
                    "borderWidth": 0,
                    "flex": 1,
                    "fontSize": 26,
                    "paddingLeft": 20,
                  },
                  Array [
                    Object {
                      "borderWidth": 0,
                      "flex": 1,
                      "fontSize": 26,
                      "paddingLeft": 20,
                    },
                    undefined,
                    Object {
                      "paddingRight": 20,
                    },
                    Object {
                      "height": 54,
                    },
                  ],
                  Object {
                    "fontFamily": undefined,
                  },
                ]
              }
              testID="url"
              underlineColorAndroid="transparent"
              validate={[Function]}
              value=""
            />
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          dispatch={[Function]}
          name="kioskIdentifier"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Kiosk ID"
          rejectResponderTermination={true}
          returnKeyType="done"
          style={
            Array [
              Object {
                "borderWidth": 0,
                "flex": 1,
                "fontSize": 26,
                "paddingLeft": 20,
              },
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#dfdfdf",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "width": "100%",
                },
                Object {
                  "flex": 0,
                  "marginTop": 15,
                },
                Object {
                  "paddingRight": 20,
                },
              ],
              Object {
                "fontFamily": undefined,
              },
            ]
          }
          testID="kioskIdentifier"
          underlineColorAndroid="transparent"
          value="1"
        />
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "marginTop": 10,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignContent": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontFamily": "Ubuntu",
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              CONNECT PRINTER
            </Text>
            <RCTSwitch
              accessibilityRole="button"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 31,
                  "width": 51,
                }
              }
              value={false}
            />
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F80347",
              "borderRadius": 5,
              "height": 75,
              "justifyContent": "center",
              "marginTop": 15,
              "opacity": 1,
            }
          }
          testID="save"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 22,
              }
            }
          >
            Save settings
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
